FileIO/GmshIO/GMSHAdaptiveMeshDensity.cpp

```diff
@@ -123,7 +123,9 @@ void GMSHAdaptiveMeshDensity::getSteinerPoints (std::vector<GeoLib::Point*> & pn
 			const double delta ((ur[0] - ll[0]) / (2 * n_pnts_per_quad_dim));
 			for (std::size_t i(0); i<n_pnts_per_quad_dim; i++) {
 				for (std::size_t j(0); j<n_pnts_per_quad_dim; j++) {
-					pnts.push_back(new GeoLib::Point (ll[0] + (2*i+1) * delta, ll[1] + (2*j+1) * delta, 0.0));
+					pnts.push_back(new GeoLib::Point(
+					    ll[0] + (2 * i + 1) * delta,
+					    ll[1] + (2 * j + 1) * delta, 0.0, pnts.size()));
```

Comment:
I guess it works, because `pnts.size()` is evaluated before `push_back()`, so it still returns the old size. Anyway, it might be clearer to introduce an auxiliary variable, like `new_point_id`, that is set before the `push_back`.